### PR TITLE
Add skill for baseline source mapping matrix

### PR DESCRIPTION
**Pull request by dhwanilpatel**
Add skill which help researching the source database and create baseline mapping/feature matrix against spanner. 
User can act on it to achieve the final mapping/feature matrix which will be used for supporting the source in SMT.

Skill will generate two files in below format.

*  (source)_datatype_mapping_matrix.csv

|Mapping Category | Source Database Type / Alias | Is Source Datatype Supported as PK? | Spanner GoogleSQL Default Datatype | Spanner GoogleSQL Default Datatype If Column is PK | Spanner GoogleSQL Alternative Datatypes | Spanner PostgreSQL Default Datatype | Spanner PostgreSQL Default Datatype If Column is PK | Spanner PostgreSQL Alternative Datatypes | Datastream Support Status | Edge Cases for Smoke Test | Notes
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

*  (source)_feature_mapping_matrix.csv

|Category | Feature Supported in Source | Spanner GoogleSQL Equivalent Feature | Notes (Spanner GoogleSQL) | Spanner PostgreSQL Equivalent Feature | Notes (Spanner PostgreSQL) | Source DB Documentation URL | Spanner Documentation URL
-- | -- | -- | -- | -- | -- | -- | --

### CODE REVIEW

File: .agents/skills/source_research_helper/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+>
+> **Anti-Summarization Rule**: You are strictly forbidden from summarizing or limiting the list to "common" migration types. You must actively search for and include the database's "long tail" data types.
+> **Do not pre-filter types**: Do not silently omit obscure, legacy, or system data types. You MUST include them in the mapping matrix and mark them as Unsupported (Scenario D) with a rationale if they are not supported in spanner. Exhaustive means exhaustive.
+
+#### 2.1 Programmatic Raw Extraction (First Principles Checklist)
+- Write and execute a script (or use available web/file tools) to scrape and extract the complete, unfiltered list of *all* native datatypes and their official aliases directly from the official database documentation.
+- **Collection & UDT Mandate**: You MUST actively search for and explicitly extract all Object-Relational Collection Types and hierarchical User-Defined Types. Do not restrict your extraction exclusively to simple scalar primitives.
```
Comment (bharadwaj-aditya) [inline .agents/skills/source_research_helper/SKILL.md:31]: We would not build migration tools for user defined types. That should not be here.

File: .agents/skills/source_research_helper/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  - JSON/XML Types
+  - Geometric/Spatial Types
+  - Network Address Types
+  - UUID/Identifier Types
+  - Range & Multirange Types
+  - User-Defined & Custom Types
+  - System/Internal Types
+
+#### 2.4 Discover Alternative Mappings in Conversion Code & Public Dialects
+- Analyze the Dataflow template's generic conversion classes (e.g., `GenericRecordTypeConvertor.java` and `AvroToValueMapper.java`) to understand what alternative mappings are supported via user overrides.
```
Comment (bharadwaj-aditya) [inline .agents/skills/source_research_helper/SKILL.md]: The alternate types should be anything that is supported, not just the list in the genericRecortType Convertor.

Also, we should provide the full link of the repo/file here here rather than just the name

File: .agents/skills/source_research_helper/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+- **Leverage Public Mapping References (Cheat Sheets):** Research how mature open-source projects or JDBC drivers map the type:
+  - **Hibernate Dialects**: Search for the target database's Hibernate Dialect class.
```
Comment (bharadwaj-aditya) [inline .agents/skills/source_research_helper/SKILL.md]: why is hibernate relevant for this use case ?

File: .agents/skills/source_research_helper/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  - **JDBC Driver Type Mappings**: Check how the JDBC driver handles the type (e.g., mapping IPADDRESS to binary format).
+  - **Beam SchemaUtil**: Check `org.apache.beam.sdk.io.jdbc.SchemaUtil` to see how Beam maps standard JDBC types.
+- For every supported type, identify **ALL viable, structurally valid alternative Spanner mappings** (Scenario B) based on this research. Do not limit to just one alternative; list all possible fallback types (e.g., `INT64`, `STRING`, `FLOAT64` for numerical fallbacks).
+
+## 3. Creating the Logical Type Mapping Table
+Construct a markdown table containing exactly these columns:
```
Comment (bharadwaj-aditya) [inline .agents/skills/source_research_helper/SKILL.md:70]: will this be a markdown or a spreadsheet ? How will someone make this a spreadsheet to review and make changes ?

File: .agents/skills/source_research_helper/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+## 4. Feature Support Discovery
+
+> [!IMPORTANT]
+> **Anti-Hallucination Rule**: You are strictly forbidden from relying on your internal LLM memory to evaluate feature support. You MUST use search tools to locate official documentation for every single feature before mapping it.
+
+### 4.1 The Master Validation List & Strict Categorization Taxonomy
+You must explicitly research and evaluate the source database for all of the critical architectural features listed below. 
+
+**Strict Category Names:** When populating the `Category` column in the matrix, you MUST strictly use exactly these 5 bolded category strings for the items listed beneath them. Downstream automated templates rigidly parse this output expecting these exact category strings. 
+However, **you are NOT restricted to only these categories.** You must research other core database functionalities (e.g., Stored Procedures, Sequences, Row-Level Security) or proprietary edge cases, invent appropriate new category names for them, and append them to the report alongside these mandatory minimums.
```
Comment (bharadwaj-aditya) [inline .agents/skills/source_research_helper/SKILL.md]: maybe just call the remaining as others ? Leaving it open may lead to category values explosion

**Review by bharadwaj-aditya — COMMENTED** [review]
Please also checkin a sample of both files for Mysql

File: .agents/skills/source_research_helper/SKILL.md (hunk 1 above)
Comment (dhwanilpatel) [reply .agents/skills/source_research_helper/SKILL.md:31]: I included them for completeness across all datatypes. The report should include these items, but they would marked as not supported.

File: .agents/skills/source_research_helper/SKILL.md (hunk 2 above)
Comment (dhwanilpatel) [reply .agents/skills/source_research_helper/SKILL.md]: > The alternate types should be anything that is supported, not just the list in the genericRecortType Convertor.

Yes, it will try to include all supported alternative types. To identify it, the skill checks multiple sources. `genericRecordConvertor` is one of these sources, we also check other locations (such as the public mapping reference) to find supported alternative types.

> Also, we should provide the full link of the repo/file here here rather than just the name

Sure, will add the link of the DataflowTemplate repo and provide the file name.

File: .agents/skills/source_research_helper/SKILL.md (hunk 3 above)
Comment (dhwanilpatel) [reply .agents/skills/source_research_helper/SKILL.md]: Intention was to provide some example of public reference to look for like hibernate/JDBC driver/etc.

File: .agents/skills/source_research_helper/SKILL.md (hunk 4 above)
Comment (dhwanilpatel) [reply .agents/skills/source_research_helper/SKILL.md:70]: > will this be a markdown or a spreadsheet ?

Internally it creates the markdown, but it has been highlighted in `Dual-Format Artifact Generation` section to generate the csv. 

> How will someone make this a spreadsheet to review and make changes ?

This would be covered in the guideline which we will create for source support.

File: .agents/skills/source_research_helper/SKILL.md (hunk 5 above)
Comment (dhwanilpatel) [reply .agents/skills/source_research_helper/SKILL.md]: Ack, will update it to map it to Others category.

File: .agents/skills/source_research_helper/SKILL.md (hunk 3 above)
Comment (bharadwaj-aditya) [reply .agents/skills/source_research_helper/SKILL.md]: I think having JDBC is fine. Let's not add references to specific ORMs

Comment (dhwanilpatel) [reply .agents/skills/source_research_helper/SKILL.md]: Ack. Removed hibernate from skill.

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM
